dask/array/gufunc.py

```diff
@@ -147,7 +174,11 @@ def apply_gufunc(func, signature, *args, **kwargs):
 
     ## Determine and handle output_dtypes
     if output_dtypes is None:
-        output_dtypes = apply_infer_dtype(func, args, kwargs, "apply_gufunc", "output_dtypes", nout)
+        if vectorize:
+            _func = np.vectorize(func, signature=signature)
```

Comment:
style nit: I would generally only preface variables with an underscore if it serves to mark a variable as private. In this case, these are already local variables so I would stick to non-prefaced names.